### PR TITLE
[skip-ci] Add documentation for SSL_*_ex_data() functions

### PR DESCRIPTION
**Pull request by jon-oracle**
##### Checklist
- [x] documentation is added or updated

##### Description of change
Add documentation for SSL_*_ex_data() functions

### CODE REVIEW

File: doc/man3/SSL_CTX_set_ex_data.pod [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ void *SSL_SESSION_get_ex_data(const SSL_SESSION *s, int idx);
+
+=head1 DESCRIPTION
+
+SSL*_set_ex_data() functions can be used to store arbitrary user data into the
+B<SSL_CTX>, B<SSL> or B<SSL_SESSION>. The user must supply a unique index
+which they can subsequently use to retrieve the data using SSL*_get_ex_data().
+
+For more detailed information see L<CRYPTO_get_ex_data(3)> and
+L<CRYPTO_set_ex_data(3)> which implement these functions.
```
Comment (levitte) [inline doc/man3/SSL_CTX_set_ex_data.pod]: That's kinda nice, but most importantly, you might want to have people look at `CRYPTO_get_ex_new_index()` for that unique index.

File: doc/man3/SSL_CTX_set_ex_data.pod [hunk 2]
```diff
@@ -0,0 +1,56 @@
+=pod
+
+=head1 NAME
+
+SSL_CTX_get_ex_data, SSL_CTX_set_ex_data,
+SSL_get_ex_data, SSL_set_ex_data,
+SSL_SESSION_get_ex_data, SSL_SESSION_set_ex_data
+- Store and retrieve extra data from the B<SSL_CTX>, B<SSL> or B<SSL_SESSION>
```
Comment (levitte) [inline doc/man3/SSL_CTX_set_ex_data.pod]: Can't have POD markud in the NAME section...

**Comment by richsalz** [comment]
Just this patch catch it?
```
; vi ./util/find-doc-nits
; g diff
diff --git a/util/find-doc-nits b/util/find-doc-nits
index f14e36e..712a398 100755
--- a/util/find-doc-nits
+++ b/util/find-doc-nits
@@ -62,6 +62,7 @@ sub name_synopsis()
     print "$id trailing comma before - in NAME\n" if $tmp =~ /, *-/;
     $tmp =~ s/-.*//g;
     $tmp =~ s/,//g;
+    print "$id POD markup in NAME\n" if $tmp =~ /[A-Z]</;

     my $dirname = dirname($filename);
     my $simplename = basename($filename);
;
```
?

**Comment by levitte** [comment]
@richsalz, not sure what you mean, there already is a check, [here](https://github.com/openssl/openssl/blob/master/util/find-doc-nits#L161-L162)

**Comment by richsalz** [comment]
"never mind."  I didn't look.  I just thought folks would run find-doc-nits first, and therefore assumed it was missing a check :)

**Comment by levitte** [comment]
I'll try to remember to be more precise and point out when things are found by Travis...

File: doc/man3/SSL_CTX_set_ex_data.pod (hunk 1 above)
Comment (jon-oracle) [inline doc/man3/SSL_CTX_set_ex_data.pod]: Sure, I'll add that

**Comment by levitte** [comment]
Merged

master: 09fdfa4b2fe4c9770c27cd523a5bf55b1d50eac9
1.1.0: fd2975883d51c475247d3d8a5919b380944b9a46

**Comment by jon-oracle** [comment]
It looks like a conflicting POD has been added elsewhere around the same time as this change:

$ ./util/find-doc-nits -n
doc/man3/SSL_CTX_set_ex_data.pod:1: the following exist as other .pod files:
SSL_SESSION_get_ex_data

**Comment by richsalz** [comment]
yeah #3068

**Pull request closed without merging**